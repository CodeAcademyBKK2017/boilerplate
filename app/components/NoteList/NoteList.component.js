import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './NoteList.style';
import {
  FlatList, Text, View
} from 'react-native';

export default class NoteList extends Component {
  renderItem = ({item}) => (
    <View style={styles.itemContainer}>
      <Text style={styles.itemTitle}>{item.title}</Text>
      <Text style={styles.itemContent}>{item.content}</Text>
    </View>
  );

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.noteTitle}>Notes:</Text>
        <FlatList
          data={this.props.data}
          renderItem={this.renderItem}/>
      </View>
    );
  }
}

NoteList.propTypes = {
  data: PropTypes.array.isRequired
};

NoteList.defaultProps = {
  data: []
};
