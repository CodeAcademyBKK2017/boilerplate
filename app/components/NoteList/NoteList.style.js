import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%'
  },
  noteTitle: {
    fontWeight: 'bold'
  },
  itemContainer: {
    paddingVertical: 10
  },
  itemTitle: {
    fontWeight: 'bold',
    fontSize: 14,
    paddingBottom: 10
  },
  itemContent: {
    fontSize: 12
  }
});

export default styles;
