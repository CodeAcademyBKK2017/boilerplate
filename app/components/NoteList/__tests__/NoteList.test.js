import React from 'react';
import 'react-native';
import NoteList from '../NoteList.component';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('NoteList', () => {
  let wrapper, instance, notes;
  beforeEach(() => {
    wrapper = shallow(<NoteList />);
    instance = wrapper.instance();
    notes = [
      {title: 'Title', content: 'Content', uuid: '1'},
      {title: 'Title 2', content: 'Content 2', uuid: '2'},
      {title: 'Title 3', content: 'Content 3', uuid: '3'}
    ];
  });
  it('renders correctly', () => {
    const snapshot = renderer.create(<NoteList notes={notes} />);
    expect(snapshot).toMatchSnapshot();
  });
  it('keyExtractor', () => {
    const value = instance.keyExtractor({title: 'Title', content: 'Content', uuid: '1'});
    expect(value).toEqual('1');
  });
  it('renderItem', () => {
    const item = {title: 'Title', content: 'Content', uuid: '1'};
    const snapshot = instance.renderItem({item});
    expect(snapshot).toMatchSnapshot();
  });
  it('showModal', () => {
    const item = {title: 'Title', content: 'Content'};
    const expectedState = {modalVisible: true, modalTitle: 'Title', modalContent: 'Content'};
    instance.showModal(item)();
    expect(instance.state).toMatchObject(expectedState);
  });
  it('onModalClose', () => {
    instance.onModalClose();
    expect(instance.state.modalVisible).toBe(false);
  });
});