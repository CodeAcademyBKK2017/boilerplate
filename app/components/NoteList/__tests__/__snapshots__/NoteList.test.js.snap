// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList renderItem 1`] = `
<Swipeout
  autoClose={true}
  disabled={false}
  right={
    Array [
      Object {
        "backgroundColor": "#003",
        "onPress": [Function],
        "text": "Info",
      },
      Object {
        "backgroundColor": "#666",
        "text": "Cancel",
      },
      Object {
        "backgroundColor": "#900",
        "onPress": undefined,
        "text": "Delete",
      },
    ]
  }
  rowID={-1}
  sectionID={-1}
  sensitivity={50}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Title
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingLeft": 10,
            }
          }
        >
          Content
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</Swipeout>
`;

exports[`NoteList renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "content": "Content",
          "title": "Title",
          "uuid": "1",
        },
        Object {
          "content": "Content 2",
          "title": "Title 2",
          "uuid": "2",
        },
        Object {
          "content": "Content 3",
          "title": "Title 3",
          "uuid": "3",
        },
      ]
    }
    disableVirtualization={false}
    extraData={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#dbddde",
        "padding": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "left": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Title
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "paddingLeft": 10,
                      }
                    }
                  >
                    Content
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "left": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Title 2
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "paddingLeft": 10,
                      }
                    }
                  >
                    Content 2
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "left": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Title 3
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "paddingLeft": 10,
                      }
                    }
                  >
                    Content 3
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationOutType="fadeOut"
    animationType="none"
    hardwareAccelerated={false}
    onClose={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(37, 8, 10, 0.78)",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "padding": 20,
        }
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "padding": 20,
            },
            undefined,
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={Object {}}
        >
          
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
